File: python/src/models/compass/cancel_query_run.py
from typing import List, Union

from pydantic import BaseModel

from .core.query_run import QueryRun
from .core.rpc_request import RpcRequest
from .core.rpc_response import RpcResponse


# Request
class CancelQueryRunRpcRequestParams(BaseModel):
    queryRunId: str


class CancelQueryRunRpcRequest(RpcRequest):
    method: str = "cancelQueryRun"
    params: List[CancelQueryRunRpcRequestParams]


# Response
class CancelQueryRunRpcResult(BaseModel):
    queryRun: QueryRun


class CancelQueryRunRpcResponse(RpcResponse):
    result: Union[CancelQueryRunRpcResult, None]

File: js/src/types/compass/cancel-query-run.ts
import { QueryRun, RpcRequest, BaseRpcRequest, RpcResponse, BaseRpcResponse } from "./core";

// Request
export interface CancelQueryRunRpcRequestParams {
  queryRunId: string;
}

export interface CancelQueryRunRpcRequest extends RpcRequest<CancelQueryRunRpcRequestParams> {
  method: "cancelQueryRun";
}

export class CancelQueryRunRpcRequestImplementation
  extends BaseRpcRequest<CancelQueryRunRpcRequestParams>
  implements CancelQueryRunRpcRequest
{
  method: "cancelQueryRun" = "cancelQueryRun";
}

// Response
export interface CancelQueryRunRpcResult {
  queryRun: QueryRun;
}

export interface CancelQueryRunRpcResponse extends RpcResponse<CancelQueryRunRpcResult> {}

export class CancelQueryRunRpcResponseImplementation
  extends BaseRpcResponse<CancelQueryRunRpcResult>
  implements CancelQueryRunRpcResponse {}
